tutorials/Tutorial7_generator.py

```diff
@@ -0,0 +1,75 @@
+from transformers import RagTokenizer, RagRetriever, RagTokenForGeneration, RagSequenceForGeneration
+import torch
+
+from haystack import Document
+from haystack.document_store.faiss import FAISSDocumentStore
+from haystack.retriever.dense import DensePassageRetriever
+
+documents = [
+    Document(
+        text="""Aaron Aaron ( or ; ""Ahärôn"") is a prophet, high priest, and the brother of Moses in the Abrahamic religions. Knowledge of Aaron, along with his brother Moses, comes exclusively from religious texts, such as the Bible and Quran. The Hebrew Bible relates that, unlike Moses, who grew up in the Egyptian royal court, Aaron and his elder sister Miriam remained with their kinsmen in the eastern border-land of Egypt (Goshen). When Moses first confronted the Egyptian king about the Israelites, Aaron served as his brother's spokesman (""prophet"") to the Pharaoh. Part of the Law (Torah) that Moses received from""",
+        meta={"name": "0"}
+    ),
+    Document(
+        text="""Democratic Republic of the Congo to the south. Angola's capital, Luanda, lies on the Atlantic coast in the northwest of the country. Angola, although located in a tropical zone, has a climate that is not characterized for this region, due to the confluence of three factors: As a result, Angola's climate is characterized by two seasons: rainfall from October to April and drought, known as ""Cacimbo"", from May to August, drier, as the name implies, and with lower temperatures. On the other hand, while the coastline has high rainfall rates, decreasing from North to South and from to , with""",
+    ),
+    Document(
+        text="""Schopenhauer, describing him as an ultimately shallow thinker: ""Schopenhauer has quite a crude mind ... where real depth starts, his comes to an end."" His friend Bertrand Russell had a low opinion on the philosopher, and attacked him in his famous ""History of Western Philosophy"" for hypocritically praising asceticism yet not acting upon it. On the opposite isle of Russell on the foundations of mathematics, the Dutch mathematician L. E. J. Brouwer incorporated the ideas of Kant and Schopenhauer in intuitionism, where mathematics is considered a purely mental activity, instead of an analytic activity wherein objective properties of reality are""",
+        meta={"name": "1"}
+    ),
+    Document(
+        text="""The Dothraki vocabulary was created by David J. Peterson well in advance of the adaptation. HBO hired the Language Creatio""",
+        meta={"name": "2"}
+    ),
+    Document(
+        text="""The title of the episode refers to the Great Sept of Baelor, the main religious building in King's Landing, where the episode's pivotal scene takes place. In the world created by George R. R. Martin""",
+        meta={}
+    )
+]
+
+document_store = FAISSDocumentStore(faiss_index_factory_str="HNSW")
```

Comment:
For a tutorial, I would go with the default flat index type (speed doesn't make a big difference below 10k docs)

Reply:
Currently this tutorial is a testing ground for me and using it to debugging. Once I will find correct way to integrate rag with haystack, I will remove all but keep proper tutorial to generator via Finder

Reply:
Trying to minimise difference with original paper.

Reply:
Ok, for testing and potential comparisons this makes sense.
Afterwards, I would still change it to `Flat` as this is the more reasonable default here that users should work with.
In a final Tutorial, I would probably add a comment there explaining the choices and in what scenarios HNSW will be beneficial